### PR TITLE
simple-git scenario tests

### PR DESCRIPTION
**Pull request by davidzwa**
~I requested these on the last PR, but got skipped. Thought as updater is super important I'd add them myself.~

- [x] Test not a git repo, and success variant
- [x] Test modifications when up-to-date, behind or ahead **with force true**
- [x] Test modifications when behind **without force**
- [x] Test behind + missing packages

Fixes:
- [x] statusTypeForUser undefined for missing packages scenario

### CODE REVIEW

**Comment by NotExpectedYet** [comment]
You approved, I smashed merge. Was still planning to add them.

**Comment by davidzwa** [comment]
Yep true that

File: test/application/server-commands.test.js [hunk 1]
```diff
@@ -1,95 +1,171 @@
...
+        const upDateCompletedMessage = "Update command has run successfully, OctoFarm will restart.";
+        const missingPackagesMessage = "You have missing dependencies that are required, Do you want to update these?";
+        const modificationsDetectedMessage = "The update is failing due to local changes been detected. Please check the file list below for what has been modified.";
 
-    it("should not reboot octofarm in unknown mode", async () => {
-        // Output indicates that we are neither in pm2 or nodemon mode
-        const output = await SystemCommands.rebootOctoFarm();
-        expect(output).toBe(false);
-    });
+        const scenarioOutcomes = [
```
Comment (davidzwa) [inline test/application/server-commands.test.js:69]: Never tried this before. Found it pretty cool to make!

File: server_src/lib/serverCommands.js [hunk 2]
```diff
@@ -46,6 +46,7 @@ class SystemCommands {
     // Check to see if current dir contains a git folder... hard fail otherwise.
     let isThisAGitRepo = await checkIfWereInAGitRepo();
     if (!isThisAGitRepo) {
+      clientResponse.statusTypeForUser = "warning";
```
Comment (davidzwa) [inline server_src/lib/serverCommands.js:49]: Missing the type.

File: server_src/lib/serverCommands.js [hunk 3]
```diff
@@ -109,6 +108,7 @@ class SystemCommands {
     // If we have missing packages alert the user and wait for their response, if response given then install missing deps.
     if (missingPackages) {
       if (!force?.doWeInstallPackages) {
+        clientResponse.statusTypeForUser = "warning";
```
Comment (davidzwa) [inline server_src/lib/serverCommands.js:111]: Missing type prop as well

**Review by NotExpectedYet — APPROVED** [review]

**Pull request merged**